iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/RunInstructionImplTest.java

```diff
@@ -315,4 +317,101 @@ void shouldCheckParseRunMultiLineMultipleRedirect() {
+    assertThat(syntaxToken.value()).isEqualTo("parameters");
+
+    assertThat(syntaxToken.comments()).hasSize(1);
+    Comment comment = syntaxToken.comments().get(0);
+    assertThat(comment.value()).isEqualTo("# my comment");
+    assertThat(comment.contentText()).isEqualTo("# my comment");
+    assertTextRange(comment.textRange()).hasRange(2,1,2,13);
+  }
+
+  @Test
+  void shouldHaveInlineCommentAttachedToNextElementSeparatedTokensMultipleComment() {
+    String toParse = "RUN executable \\\n" +
+      "# my comment 1\n" +
+      "  # my comment 2\n" +
+      "      parameters";
+    RunInstruction tree = DockerTestUtils.parse(toParse, DockerLexicalGrammar.RUN);
+    assertTextRange(tree.textRange()).hasRange(1,0,4,16);
+
+    assertThat(tree.keyword().value()).isEqualTo("RUN");
+    assertThat(tree.arguments()).hasSize(2);
+    assertArgumentsValue(tree.arguments(), "executable", "parameters");
+
+    SyntaxTokenImpl syntaxToken = (SyntaxTokenImpl) TreeUtils.getLastDescendant(tree, SyntaxTokenImpl.class::isInstance).get();
+    assertThat(syntaxToken.value()).isEqualTo("parameters");
+    assertThat(syntaxToken.comments())
+      .hasSize(2)
+      .extracting("value").containsExactly("# my comment 1", "# my comment 2");
+  }
+
+  @Test
+  void shouldHaveInlineCommentAttachedToNextElementMergedTokens() {
+    String toParse = "RUN executable\\\n" +
+      "# my comment\n" +
+      "parameters";
+    RunInstruction tree = DockerTestUtils.parse(toParse, DockerLexicalGrammar.RUN);
+    assertTextRange(tree.textRange()).hasRange(1,0,3,10);
+
+    assertThat(tree.keyword().value()).isEqualTo("RUN");
+    assertThat(tree.arguments()).hasSize(1);
+    assertArgumentsValue(tree.arguments(), "executableparameters");
+
+    SyntaxTokenImpl syntaxToken = (SyntaxTokenImpl) TreeUtils.getLastDescendant(tree, SyntaxTokenImpl.class::isInstance).get();
+    assertThat(syntaxToken.value()).isEqualTo("executableparameters");
+    assertThat(syntaxToken.comments())
+      .hasSize(1)
+      .extracting("value").containsExactly("# my comment");
+  }
+
+  @Test
+  void shouldHaveInlineCommentAttachedToNextElementMergedTokensFinishingAtNextLine() {
+    String toParse = "RUN executable\\\n" +
+      "# my comment\n" +
+      "parameters\\\n" +
+      " parameter2";
+    RunInstruction tree = DockerTestUtils.parse(toParse, DockerLexicalGrammar.RUN);
+    assertTextRange(tree.textRange()).hasRange(1,0,4,11);
+
+    assertThat(tree.keyword().value()).isEqualTo("RUN");
+    assertThat(tree.arguments()).hasSize(2);
+    assertArgumentsValue(tree.arguments(), "executableparameters", "parameter2");
+
+    SyntaxTokenImpl syntaxToken = (SyntaxTokenImpl) TreeUtils.getLastDescendant(tree, SyntaxTokenImpl.class::isInstance).get();
+    assertThat(syntaxToken.value()).isEqualTo("parameter2");
+    assertThat(syntaxToken.comments())
+      .hasSize(1)
+      .extracting("value").containsExactly("# my comment");
+  }
```

Comment:
Couldn't fix this : the inline comment get attached to the `SyntaxToken` `"parameter2"` instead of the merged `SyntaxToken` `"executableparameters"`.
This is because to prevent attaching the comment to the previous `SyntaxToken`, I added a check to detect when overlapping parent of the comment end at offset `0` (which is the current behavior of our elements at ends of lines).
But unfortunately this check has the side effect that we also exclude overlapping parent which are continued and last the whole next line(s) (probably a very rare case).